openwisp_controller/config/base/device.py

```diff
@@ -261,6 +261,7 @@ def _validate_unique_name(self):
                 )
 
     def clean(self, *args, **kwargs):
+        self.mac_address = self.mac_address.upper()
         super().clean(*args, **kwargs)
```

Comment:
I believe calling `super().clean()` first is safer:

```suggestion
        super().clean(*args, **kwargs)
        self.mac_address = self.mac_address.upper()
```